### PR TITLE
refactor(lib): `NewHTTPRequestHandler` no longer relies on `UnmarshalParams`

### LINKED ISSUE #1750: `NewHTTPRequestHandler` should use `json.Decode` to fill params
Right now, `NewHTTPRequestHandler` calls `UnmarshalParams`. This splits logic depending on the request header and the content type.

The lib level should only be expecting http POST responses with json bodies, any other type of requests should be handled as syntactic sugar at the api level.

This issue is closed when:
- [x] remove `UnmarshalParams` in `NewHTTPRequestHandler` in favor of decoding json
- [x] examine where else `UnmarshalParams` is being used
- [x] examine where the `Unmarshaller` interface is being used 
- [x] examine which, if any Params are using their UnmarshalFromRequest. The main suspect here would be `lib.Get` aka `/get`
- [x] remove any unused code

### PR DESCRIPTION
**Pull request by ramfox**
closes #1750 

- `NewHTTPRequestHandler` relies on new `DecodeParams` func, that expects to decode the request body as json
- moves `Unmarshaller` interface and `UnmarshalParams` to new `api/params.go` file
- removes unused `UnmarshalFromParams` methods
- adds `api/handlers.go` and moves `GetHandler` & `UnpackHandler` to this file
- removes the `DatasetHandlers` in favor of passing a `Instance` to `GetHandler` directly
- adds functionality to `api` test runner to clean up any unused `api.snapshots`

### CODE REVIEW

File: api/handlers.go [hunk 1]
```diff
@@ -16,34 +16,21 @@ import (
...
-
-// NewDatasetHandlers allocates a DatasetHandlers pointer
-func NewDatasetHandlers(inst *lib.Instance, readOnly bool) *DatasetHandlers {
-	h := DatasetHandlers{inst.Dataset(), inst, readOnly}
-	return &h
-}
-
 // GetHandler is a dataset single endpoint
-func (h *DatasetHandlers) GetHandler(routePrefix string) http.HandlerFunc {
+func GetHandler(inst *lib.Instance, routePrefix string) http.HandlerFunc {
```
Comment (ramfox) [inline api/handlers.go:20]: probably most controversial thing in this pr, removing `DatasetHandlers` and passing an `inst` directly into the `GetHandler`

File: lib/fsi.go [hunk 2]
```diff
@@ -45,14 +44,7 @@ func (m FSIMethods) Attributes() map[string]AttributeSet {
...
 	Dir    string `qri:"fspath"`
-	Refstr string
-}
-
-// UnmarshalFromRequest implements a custom deserialization-from-HTTP request
-func (p *LinkParams) UnmarshalFromRequest(r *http.Request) error {
-	p.Refstr = r.FormValue("refstr")
-	p.Dir = r.FormValue("dir")
-	return nil
+	Refstr string `json:"refstr"`
```
Comment (ramfox) [inline lib/fsi.go:47]: Once this is rebased to #1754 this won't be a relevant change.

File: api/params.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	// should be removed and should error out on unknown keys
+	decoder.IgnoreUnknownKeys(true)
+}
+
+// RequestUnmarshaller is an interface for deserializing from an HTTP request
+type RequestUnmarshaller interface {
+	UnmarshalFromRequest(r *http.Request) error
+}
+
+// UnmarshalParams deserialzes a lib req params stuct pointer from an HTTP
```
Comment (dustmop) [inline api/params.go:24]: deserialzes -> deserializes

File: api/handlers.go (hunk 1 above)
Comment (dustmop) [reply api/handlers.go:20]: I'm totally fine with it.

**Review by dustmop — APPROVED** [review]
LGTM, just one small comment fix please

**Pull request merged**